src/main/java/com/iexec/common/chain/IexecHubAbstractService.java

```diff
@@ -913,8 +888,8 @@ public String getTaskResults(String chainTaskId, Integer chainId) {
                     .viewTaskABILegacy(BytesUtils.stringToBytes(chainTaskId)).send());
             String chainDealId = chainTask.getDealid();
             if (!StringUtils.isEmpty(chainDealId)
-                    && BytesUtils.isBytes32(BytesUtils.stringToBytes(chainDealId))
-                    && !BytesUtils.EMPTY_HEXASTRING_64.equals(chainDealId)){
+                    && BytesUtils.isBytes32(chainDealId)
```

Comment:
Thanks for the update which is right, how about `isNonZeroedBytes32` ?

Reply:
Sure, much cleaner. Thanks!